core/src/main/java/com/linecorp/armeria/server/annotation/JacksonResponseConverterFunction.java

```diff
@@ -67,13 +70,17 @@ public HttpResponse convertResponse(ServiceRequestContext ctx,
                 // Convert the object only if the charset supports UTF-8,
                 // because ObjectMapper always writes JSON document as UTF-8.
                 if (charset.contains(StandardCharsets.UTF_8)) {
-                    return HttpResponse.of(HttpStatus.OK, mediaType.withCharset(StandardCharsets.UTF_8),
-                                           mapper.writeValueAsBytes(result));
+                    final HttpData body = HttpData.of(mapper.writeValueAsBytes(result));
+                    return HttpResponse.of(headers, body, trailingHeaders);
                 }
             }
         } else if (result instanceof JsonNode) {
             // No media type is specified, but the result is a JsonNode type.
-            return HttpResponse.of(HttpStatus.OK, MediaType.JSON_UTF_8, mapper.writeValueAsBytes(result));
+            final HttpHeaders responseHeaders =
+                    headers.isImmutable() ? HttpHeaders.copyOf(headers).contentType(MediaType.JSON_UTF_8)
+                                          : headers.contentType(MediaType.JSON_UTF_8);
+            final HttpData body = HttpData.of(mapper.writeValueAsBytes(result));
+            return HttpResponse.of(responseHeaders, body, trailingHeaders);
```

Comment:
I see some inconsistency in variable/parameter naming - `body` vs. `content`.

Reply:
Oh, I missed fixing here. Thanks!